src/helpers.ts

```diff
@@ -3,14 +3,19 @@ import type { ErrorLike } from './types';
 export function checkValidArgs(keyValuePairs: unknown[], callback: unknown) {
   if (
     !Array.isArray(keyValuePairs) ||
-    keyValuePairs.length === 0 ||
-    !Array.isArray(keyValuePairs[0])
+    (keyValuePairs.length > 0 && !Array.isArray(keyValuePairs[0]))
```

Comment:
not sure i understand this condition

Reply:
It preserves the check that verifies the first argument is also an array. E.g. `['test']` would fail and `[['key', 'val']]`  would pass.